--- evolution-3.1.92/configure.ac.wiget	2011-09-20 12:46:18.597079039 +0200
+++ evolution-3.1.92/configure.ac	2011-09-20 12:46:56.003114171 +0200
@@ -50,7 +50,7 @@ m4_define([libpst_minimum_version], [0.6
 m4_define([libnotify_minimum_version], [0.5.1])
 
 dnl Optional Packages
-m4_define([champlain_minimum_version], [0.10])
+m4_define([champlain_minimum_version], [0.12])
 m4_define([clutter_gtk_minimum_version], [0.90])
 m4_define([geoclue_minimum_version], [0.11.1])
 m4_define([gladeui_minimum_version], [3.10.0])
@@ -1500,12 +1500,12 @@ if test "x$enable_contact_maps" = "xyes"
 		AC_MSG_ERROR([Clutter is required for maps in contacts. Use --with-clutter=yes to enable clutter.])
 	fi
 
-	PKG_CHECK_MODULES([CHAMPLAIN], [champlain-gtk-0.10 >= champlain_minimum_version], [have_champlain=yes], [have_champlain=no])
+	PKG_CHECK_MODULES([CHAMPLAIN], [champlain-gtk-0.12 >= champlain_minimum_version], [have_champlain=yes], [have_champlain=no])
 	AC_SUBST(CHAMPLAIN_CFLAGS)
 	AC_SUBST(CHAMPLAIN_LIBS)
 
 	if test "x$have_champlain" = "xno"; then
-		AC_MSG_ERROR([champlain-gtk-0.10 >= champlain_minimum_version is required for maps in contacts preview.])
+		AC_MSG_ERROR([champlain-gtk-0.12 >= champlain_minimum_version is required for maps in contacts preview.])
 	fi
 
 	PKG_CHECK_MODULES([GEOCLUE], [geoclue >= geoclue_minimum_version], [have_geoclue=yes], [have_geoclue=no])
--- evolution-3.1.92/widgets/misc/e-contact-map.c.wiget	2011-09-20 13:04:10.735525195 +0200
+++ evolution-3.1.92/widgets/misc/e-contact-map.c	2011-09-20 13:04:33.191856685 +0200
@@ -42,7 +42,7 @@
 #include <glib/gi18n.h>
 #include <math.h>
 
-G_DEFINE_TYPE (EContactMap, e_contact_map, GTK_TYPE_CHAMPLAIN_EMBED)
+G_DEFINE_TYPE (EContactMap, e_contact_map, GTK_CHAMPLAIN_TYPE_EMBED)
 
 struct _EContactMapPrivate {
 	GHashTable *markers; /* Hash table contact-name -> marker */
