--- evolution-1.4.5/camel/Makefile.am.orig	2003-06-12 17:03:35.000000000 +0000
+++ evolution-1.4.5/camel/Makefile.am	2003-11-23 13:18:40.000000000 +0000
@@ -8,7 +8,6 @@
 camellibexec_PROGRAMS =	camel-lock-helper camel-index-control
 
 privlib_LTLIBRARIES = libcamel.la
-noinst_LTLIBRARIES = libcamel-static.la
 
 INCLUDES = -I.. -I$(srcdir)/.. 			 	\
 	-I$(top_srcdir)/intl			 	\
@@ -229,10 +228,6 @@
 	$(top_builddir)/e-util/libeutil.la 	\
 	$(CAMEL_LIBS)
 
-libcamel_static_la_SOURCES = $(libcamel_la_SOURCES)
-libcamel_static_la_LIBADD  = $(libcamel_la_LIBADD)
-libcamel_static_la_LDFLAGS = --all-static
-
 camel_lock_helper_SOURCES = 			\
 	camel-lock-helper.c 			\
 	camel-lock-helper.h
--- evolution-1.4.5/libwombat/Makefile.am.orig	2003-06-23 14:00:22.000000000 +0000
+++ evolution-1.4.5/libwombat/Makefile.am	2003-11-23 13:20:45.000000000 +0000
@@ -27,13 +27,6 @@
 	wombat-client.c \
 	$(wombatinclude_HEADERS)
 
-# Static lib for pilot conduits
-
-noinst_LTLIBRARIES = libwombat-static.la
-
-libwombat_static_la_SOURCES = $(libwombat_la_SOURCES)
-libwombat_static_la_LDFLAGS = --all-static
-
 BUILT_SOURCES = $(IDL_GENERATED)
 CLEANFILES = $(BUILT_SOURCES)
 
--- evolution-1.4.5/libical/src/libical/Makefile.am.orig	2003-02-10 15:53:20.000000000 +0000
+++ evolution-1.4.5/libical/src/libical/Makefile.am	2003-11-23 13:21:43.000000000 +0000
@@ -28,14 +28,6 @@
 
 privlib_LTLIBRARIES = libical-evolution.la
 
-noinst_LTLIBRARIES = libical-static.la
-libical_static_la_SOURCES = $(libical_evolution_la_SOURCES)
-libical_static_la_LDFLAGS = --all-static
-
-# Make sure libical-evolution.la and libical-static.la are not built in
-# parallel when using a parallel make; libtool can't handle that.
-libical_static_la_DEPENDENCIES = libical-evolution.la
-
 YFLAGS = -d -v -t -pical_yy
 LFLAGS = -Pical_yy
 LEX_OUTPUT_ROOT = lex.ical_yy
--- evolution-1.4.5/e-util/ename/Makefile.am.orig	2003-02-06 21:59:58.000000000 +0000
+++ evolution-1.4.5/e-util/ename/Makefile.am	2003-11-23 13:23:09.000000000 +0000
@@ -26,11 +26,6 @@
 	e-name-western.h
 
 
-noinst_LTLIBRARIES = libename-static.la
-libename_static_la_SOURCES = $(libename_la_SOURCES)
-libename_static_la_LDFLAGS = --all-static
-
-
 noinst_PROGRAMS =		\
 	test-ename-western	\
 	test-ename-western-gtk
--- evolution-1.4.5/e-util/Makefile.am.orig	2003-09-11 16:19:50.000000000 +0000
+++ evolution-1.4.5/e-util/Makefile.am	2003-11-23 13:23:41.000000000 +0000
@@ -16,7 +16,6 @@
 	$(E_UTIL_CFLAGS)
 
 privlib_LTLIBRARIES = libeutil.la libedb3util.la libeconduit.la
-noinst_LTLIBRARIES = libeutil-static.la
 
 eutilinclude_HEADERS = 				\
 	e-account-list.h			\
@@ -107,13 +106,6 @@
 
 libeutil_la_LIBADD = $(E_UTIL_LIBS)
 
-libeutil_static_la_SOURCES = $(libeutil_la_SOURCES)
-libeutil_static_la_LIBADD  = $(libeutil_la_LIBADD)
-libeutil_static_la_LDFLAGS = -all-static
-# Make sure libeutil.la and libeutil-static.la are not built in
-# parallel when using a parallel make; libtool can't handle that.
-libeutil_static_la_DEPENDENCIES = libeutil.la
-
 econdinclude_HEADERS =		\
 	e-pilot-map.h		\
 	e-pilot-settings.h	\
--- evolution-1.4.5/calendar/cal-util/Makefile.am.orig	2003-02-24 14:33:38.000000000 +0000
+++ evolution-1.4.5/calendar/cal-util/Makefile.am	2003-11-23 13:24:15.000000000 +0000
@@ -15,7 +15,7 @@
 # cal util library
 #
 
-privlib_LTLIBRARIES = libcal-util-static.la libcal-util.la
+privlib_LTLIBRARIES = libcal-util.la
 
 libcal_util_la_SOURCES = 	\
 	cal-util-marshal.c	\
@@ -52,13 +52,6 @@
 
 BUILT_SOURCES = $(MARSHAL_GENERATED)
 
-#
-# static library for use in conduits' shared libraries
-#
-#lib_LTLIBRARIES = libcal-util-static.la
-libcal_util_static_la_SOURCES = $(libcal_util_la_SOURCES)
-libcal_util_static_la_LDFLAGS = -all-static
-
 test_recur_SOURCES = 	\
 	 test-recur.c
 
@@ -67,4 +60,4 @@
 	 $(EVOLUTION_CALENDAR_LIBS)
 
 EXTRA_DIST = \
-	cal-util-marshal.list
\ No newline at end of file
+	cal-util-marshal.list
--- evolution-1.4.5/calendar/conduits/todo/Makefile.am.orig	2003-09-11 16:19:06.000000000 +0000
+++ evolution-1.4.5/calendar/conduits/todo/Makefile.am	2003-11-23 13:27:00.000000000 +0000
@@ -18,11 +18,11 @@
 libetodo_conduit_la_LDFLAGS = -module -avoid-version
 libetodo_conduit_la_LIBADD = 						\
 	$(top_builddir)/e-util/libeutil.la				\
-	$(top_builddir)/calendar/cal-client/libcal-client-static.la	\
-	$(top_builddir)/calendar/cal-util/libcal-util-static.la		\
+	$(top_builddir)/calendar/cal-client/libcal-client.la		\
+	$(top_builddir)/calendar/cal-util/libcal-util.la		\
 	$(top_builddir)/libversit/libversit.la				\
-	$(top_builddir)/libical/src/libical/libical-static.la		\
-	$(top_builddir)/libwombat/libwombat-static.la                   \
+	$(top_builddir)/libical/src/libical/libical-evolution.la	\
+	$(top_builddir)/libwombat/libwombat.la				\
 	$(top_builddir)/e-util/libeconduit.la		 		\
 	$(EVOLUTION_CALENDAR_CONDUIT_LIBS)
 
--- evolution-1.4.5/calendar/conduits/calendar/Makefile.am.orig	2003-06-23 13:59:48.000000000 +0000
+++ evolution-1.4.5/calendar/conduits/calendar/Makefile.am	2003-11-23 13:28:24.000000000 +0000
@@ -18,11 +18,11 @@
 libecalendar_conduit_la_LDFLAGS = -module -avoid-version
 libecalendar_conduit_la_LIBADD = 					\
 	$(top_builddir)/e-util/libeutil.la				\
-	$(top_builddir)/calendar/cal-client/libcal-client-static.la	\
-	$(top_builddir)/calendar/cal-util/libcal-util-static.la		\
+	$(top_builddir)/calendar/cal-client/libcal-client.la		\
+	$(top_builddir)/calendar/cal-util/libcal-util.la		\
 	$(top_builddir)/libversit/libversit.la				\
-	$(top_builddir)/libical/src/libical/libical-static.la		\
-	$(top_builddir)/libwombat/libwombat-static.la		 	\
+	$(top_builddir)/libical/src/libical/libical-evolution.la	\
+	$(top_builddir)/libwombat/libwombat.la				\
 	$(top_builddir)/e-util/libeconduit.la	 			\
 	$(EVOLUTION_CALENDAR_CONDUIT_LIBS)
 
--- evolution-1.4.5/calendar/cal-client/Makefile.am.orig	2003-02-24 14:33:38.000000000 +0000
+++ evolution-1.4.5/calendar/cal-client/Makefile.am	2003-11-23 13:24:59.000000000 +0000
@@ -62,14 +62,6 @@
 	cal-query.h
 
 #
-# make a static library for use by calendar conduit's shared library
-#
-noinst_LTLIBRARIES = libcal-client-static.la
-libcal_client_static_la_SOURCES = $(libcal_client_la_SOURCES)
-libcal_client_static_la_LDFLAGS = --all-static
-
-
-#
 # client-test program
 #
 
--- evolution-1.4.5/addressbook/conduit/Makefile.am.orig	2003-09-11 16:18:55.000000000 +0000
+++ evolution-1.4.5/addressbook/conduit/Makefile.am	2003-11-23 13:29:00.000000000 +0000
@@ -18,8 +18,8 @@
 libeaddress_conduit_la_LDFLAGS = -module -avoid-version
 libeaddress_conduit_la_LIBADD = 					\
 	$(top_builddir)/e-util/libeutil.la				\
-	$(top_builddir)/addressbook/backend/ebook/libebook-static.la	\
-	$(top_builddir)/e-util/ename/libename-static.la			\
+	$(top_builddir)/addressbook/backend/ebook/libebook.la		\
+	$(top_builddir)/e-util/ename/libename.la			\
 	$(top_builddir)/libversit/libversit.la				\
 	$(top_builddir)/e-util/libeconduit.la		 		\
 	$(top_builddir)/camel/libcamel.la	 			\
--- evolution-1.4.5/addressbook/backend/ebook/Makefile.am.orig	2003-06-12 17:02:46.000000000 +0000
+++ evolution-1.4.5/addressbook/backend/ebook/Makefile.am	2003-11-23 13:25:34.000000000 +0000
@@ -49,11 +49,7 @@
 	-DLIBGNOME_DISABLE_DEPRECATED			\
         $(EVOLUTION_ADDRESSBOOK_CFLAGS)
 
-privlib_LTLIBRARIES = libebook.la libebook-static.la
-
-# Make sure the libraries are not built in parallel when using a parallel make;
-# libtool can't handle that.
-libebook_static_la_DEPENDENCIES = libebook.la
+privlib_LTLIBRARIES = libebook.la
 
 libebook_la_SOURCES =					\
 	$(CORBA_SOURCE)					\
@@ -93,12 +89,6 @@
 	addressbook.h
 
 
-#
-# make a static library for use by addressbook's conduit's shared library
-#
-libebook_static_la_SOURCES = $(libebook_la_SOURCES)
-libebook_static_la_LDFLAGS = -all-static
-
 # GLib marshalling cruft
 
 e-book-marshal.h: e-book-marshal.list
