diff -urN evolution-2.28.0/m4/evo_ldap_check.m4 evolution-2.28.0.new/m4/evo_ldap_check.m4
--- evolution-2.28.0/m4/evo_ldap_check.m4	2009-08-11 13:50:27.000000000 +0200
+++ evolution-2.28.0.new/m4/evo_ldap_check.m4	2009-10-15 15:01:56.206824013 +0200
@@ -55,9 +55,9 @@
 			AC_MSG_ERROR([evolution requires OpenLDAP version >= 2])
 		fi
 
-		AC_CHECK_LIB(resolv, res_query, [LDAP_LIBS="-lresolv"])
-		AC_CHECK_LIB(socket, bind, [LDAP_LIBS="$LDAP_LIBS -lsocket"])
-		AC_CHECK_LIB(nsl, gethostbyaddr, [LDAP_LIBS="$LDAP_LIBS -lnsl"])
+		dnl AC_CHECK_LIB(resolv, res_query, [LDAP_LIBS="-lresolv"])
+		AC_CHECK_FUNC(bind, , [AC_CHECK_LIB(socket, bind, LDAP_LIBS="$LDAP_LIBS -lsocket")])
+		AC_CHECK_FUNC(gethostbyaddr, , [AC_CHECK_LIB(nsl, gethostbyaddr, LDAP_LIBS="$LDAP_LIBS -lnsl")])
 		AC_CHECK_LIB(lber, ber_get_tag, [
 			if test "$with_static_ldap" = "yes"; then
 				LDAP_LIBS="$with_openldap/lib/liblber.a $LDAP_LIBS"
