diff -urN evolution.orig/addressbook/gui/contact-editor/e-contact-editor.c evolution/addressbook/gui/contact-editor/e-contact-editor.c
--- evolution.orig/addressbook/gui/contact-editor/e-contact-editor.c	2004-06-02 12:02:59.915611392 +0200
+++ evolution/addressbook/gui/contact-editor/e-contact-editor.c	2004-06-02 12:11:13.388592096 +0200
@@ -178,6 +178,7 @@
 	{ E_CONTACT_IM_AIM,       N_ ("AIM")       },
 	{ E_CONTACT_IM_JABBER,    N_ ("Jabber")    },
 	{ E_CONTACT_IM_YAHOO,     N_ ("Yahoo")     },
+	{ E_CONTACT_IM_GADUGADU,  N_ ("Gadu-Gadu") },
 	{ E_CONTACT_IM_MSN,       N_ ("MSN")       },
 	{ E_CONTACT_IM_ICQ,       N_ ("ICQ")       },
 	{ E_CONTACT_IM_GROUPWISE, N_ ("GroupWise") }
diff -urN evolution.orig/addressbook/gui/contact-editor/e-contact-editor-im.c evolution/addressbook/gui/contact-editor/e-contact-editor-im.c
--- evolution.orig/addressbook/gui/contact-editor/e-contact-editor-im.c	2004-06-02 12:02:59.890615192 +0200
+++ evolution/addressbook/gui/contact-editor/e-contact-editor-im.c	2004-06-02 12:10:24.371043896 +0200
@@ -63,6 +63,7 @@
         N_("Novell Groupwise"),
 	N_("Jabber"),
 	N_("Yahoo Messenger"),
+	N_("Gadu-Gadu Messenger"),
 	N_("MSN Messenger"),
 	N_("ICQ")
 };
@@ -72,6 +73,7 @@
 	"im-nov",
 	"im-jabber",
 	"im-yahoo",
+	"im-gadugadu",
 	"im-msn",
 	"im-icq"
 };
diff -urN evolution.orig/addressbook/gui/widgets/eab-contact-display.c evolution/addressbook/gui/widgets/eab-contact-display.c
--- evolution.orig/addressbook/gui/widgets/eab-contact-display.c	2004-06-02 12:03:00.574511224 +0200
+++ evolution/addressbook/gui/widgets/eab-contact-display.c	2004-06-02 12:09:01.088704736 +0200
@@ -55,6 +55,7 @@
 #define JABBER_ICON       "im-jabber"
 #define MSN_ICON          "im-msn"
 #define YAHOO_ICON        "im-yahoo"
+#define GADUGADU_ICON	   "im-gadugadu"
 #define VIDEOCONF_ICON    "stock_video-conferencing"
 
 #define MAX_COMPACT_IMAGE_DIMENSION 48
@@ -356,6 +358,7 @@
 	accum_multival_attribute (accum, contact, _("Jabber"), E_CONTACT_IM_JABBER, JABBER_ICON, 0);
 	accum_multival_attribute (accum, contact, _("MSN"), E_CONTACT_IM_MSN, MSN_ICON, 0);
 	accum_multival_attribute (accum, contact, _("Yahoo"), E_CONTACT_IM_YAHOO, YAHOO_ICON, 0);
+	accum_multival_attribute (accum, contact, _("Gadu-Gadu"), E_CONTACT_IM_GADUGADU_HOME_1, GADUGADU_ICON, 0);
 
 	if (accum->len > 0)
		gtk_html_stream_printf (html_stream, accum->str);
