diff -urN aaa/evolution-1.3.0.99/camel/Makefile.am evolution2-1.3.0.99/camel/Makefile.am
--- aaa/evolution-1.3.0.99/camel/Makefile.am	Fri Feb 14 13:01:36 2003
+++ evolution2-1.3.0.99/camel/Makefile.am	Sat Feb 15 22:11:33 2003
@@ -234,12 +234,12 @@
 camel_lock_helper_SOURCES = 			\
 	camel-lock-helper.c 			\
 	camel-lock-helper.h			\
-	camel-lock.c				\
 	camel-lock.h
 
 # NOTE: This needs to be cleaned up, it shouldn't link with all this crap,
 # because it might be setuid.
 camel_lock_helper_LDADD =			\
+	camel-lock.o			\
 	$(libcamel_la_LIBADD)
 
 camel_index_control_SOURCES =			\
