diff -aurN evolution-2.3.4.orig/data/evolution.desktop.in.in evolution-2.3.4/data/evolution.desktop.in.in
--- evolution-2.3.4.orig/data/evolution.desktop.in.in	2005-06-23 08:12:19.000000000 +0200
+++ evolution-2.3.4/data/evolution.desktop.in.in	2005-07-04 20:04:55.596986296 +0200
@@ -6,7 +6,7 @@
 Icon=evolution
 Terminal=false
 Type=Application
-Categories=GNOME;Application;Office;X-Red-Hat-Base;
+Categories=GTK;GNOME;Office;
 StartupNotify=true
 X-GNOME-Bugzilla-Product=Evolution
 X-GNOME-Bugzilla-OtherBinaries=evolution-data-server-1.4;evolution-exchange-storage;evolution-alarm-notify;
