--- a/mail/em-format-html-print.c	2009/02/03 06:19:45	37216
+++ b/mail/em-format-html-print.c	2009/02/03 08:39:35	37217
@@ -71,6 +71,7 @@
 	gtk_container_add (GTK_CONTAINER (efhp->window), html);
 	gtk_widget_realize (html);
 	efhp->parent.show_icon = FALSE;
+	((EMFormat *)efhp)->print = TRUE;
 }
 
 GType
--- a/mail/em-format-html.c	2009/02/03 06:19:45	37216
+++ b/mail/em-format-html.c	2009/02/03 08:39:35	37217
@@ -1268,7 +1268,7 @@
 		if (handle)
 			handle->handler((EMFormat *)m->format, (CamelStream *)m->estream, (CamelMimePart *)m->message, handle);
 		handle = em_format_find_handler((EMFormat *)m->format, "x-evolution/message/post-header-closure");
-		if (handle)
+		if (handle && !((EMFormat *)m->format)->print)
 			handle->handler((EMFormat *)m->format, (CamelStream *)m->estream, (CamelMimePart *)m->message, handle);
 
 	}
--- a/mail/em-format.c	2009/02/03 06:19:45	37216
+++ b/mail/em-format.c	2009/02/03 08:39:35	37217
@@ -120,6 +120,7 @@
 		(GDestroyNotify) NULL,
 		(GDestroyNotify) emf_free_cache);
 	emf->composer = FALSE;
+	emf->print = FALSE;
 	emf->show_photo = TRUE;
 	emf->photo_local = TRUE;
 	e_dlist_init(&emf->header_list);
--- a/mail/em-format.h	2009/02/03 06:19:45	37216
+++ b/mail/em-format.h	2009/02/03 08:39:35	37217
@@ -228,6 +228,7 @@
 	char *charset;		/* charset override */
 	char *default_charset;	/* charset fallback */
 	gboolean composer; /* Formatting from composer ?*/
+	gboolean print;
 	gboolean show_photo; /* Want to show the photo of the sender ?*/
 	gboolean photo_local; /* Photos only from local addressbooks */
 };
