diff -urN evolution-2.0.2.orig/data/evolution.desktop.in.in evolution-2.0.2/data/evolution.desktop.in.in
--- evolution-2.0.2.orig/data/evolution.desktop.in.in	2004-11-21 15:50:42.000000000 +0100
+++ evolution-2.0.2/data/evolution.desktop.in.in	2004-11-22 23:06:50.000000000 +0100
@@ -6,9 +6,10 @@
 Icon=evolution
 Terminal=false
 Type=Application
-Categories=GNOME;Application;Office;X-Red-Hat-Base;
+Categories=GTK;GNOME;Office;
 StartupNotify=true
 X-GNOME-Bugzilla-Bugzilla=Ximian
 X-GNOME-Bugzilla-Product=Evolution
 X-GNOME-Bugzilla-OtherBinaries=evolution-data-server-1.0;evolution-exchange-storage;evolution-alarm-notify;
-
+Encoding=UTF-8
+# vi: encoding=utf-8 
