diff -urN aaa/evolution2-1.3.2.99/data/evolution.desktop.in evolution2-1.3.2.99/data/evolution.desktop.in
--- aaa/evolution2-1.3.2.99/data/evolution.desktop.in.in	Sat Mar 29 23:49:09 2003
+++ evolution2-1.3.2.99/data/evolution.desktop.in.in	Fri Apr 18 23:29:14 2003
@@ -1,6 +1,8 @@
 [Desktop Entry]
-_Name=Ximian Evolution
-_Comment=The Evolution groupware suite
+Name=Ximian Evolution
+Name[pl]=Ximian Evolution
+Comment=The Evolution groupware suite
+Comment[pl]=Zestaw programów do pracy grupowej Evolution
 Exec=evolution-@BASE_VERSION@
 Icon=evolution-@BASE_VERSION@.png
 Terminal=false
@@ -10,3 +12,5 @@
 X-GNOME-Bugzilla-Bugzilla=Ximian
 X-GNOME-Bugzilla-Product=evolution
 X-GNOME-Bugzilla-OtherBinaries=evolution-wombat;evolution-exchange-storage;
+Encoding=UTF-8
+# vi: encoding=utf-8
