diff -Nur evolution-1.2.0.orig/help/C/evolution-C.omf evolution-1.2.0.chng/help/C/evolution-C.omf
--- evolution-1.2.0.orig/help/C/evolution-C.omf	Tue Sep 24 17:45:25 2002
+++ evolution-1.2.0.chng/help/C/evolution-C.omf	Thu Nov 14 22:17:31 2002
@@ -1,16 +1,22 @@
 <?xml version="1.0" standalone="no"?>
+<!DOCTYPE omf PUBLIC "-//OMF//DTD Scrollkeeper OMF Variant V1.0//EN" "http://scrollkeeper.sourceforge.net/dtds/scrollkeeper-omf-1.0/scrollkeeper-omf.dtd">
 <omf>
   <resource>
-   <creator>
-     Ximian, Inc.
-   </creator>
+    <creator>
+      Ximian, Inc.
+    </creator>
     <title>
       Evolution Manual
     </title>
     <date>2002</date>
-    <subject category="GNOME|Applications"/>
-    <format mime="text/html"/>
+    <version identifier="" date="2002-10-03"/>
+    <subject category="GNOME|Applikasjoner"/>
+    <description/>
+    <type/>
+    <format mime="text/sgml"/>
     <identifier url="index.html"/>
     <language code="C"/>
+    <relation seriesid="f98af356-d6f5-11d6-94dd-a55e7b20f7b9"/>
+    <rights type="GNU GPL" license.version="2" license="http://www.gnu.org/licenses/gpl.html"/>
   </resource>
 </omf>
diff -Nur evolution-1.2.0.orig/help/no/evolution-no.omf evolution-1.2.0.chng/help/no/evolution-no.omf
--- evolution-1.2.0.orig/help/no/evolution-no.omf	Tue Sep 24 17:45:27 2002
+++ evolution-1.2.0.chng/help/no/evolution-no.omf	Thu Nov 14 22:20:38 2002
@@ -1,16 +1,22 @@
-<?xml version="1.0" standalone="no"?>
+<?xml version="1.0" encoding="ISO-8859-13" standalone="no"?>
+<!DOCTYPE omf PUBLIC "-//OMF//DTD Scrollkeeper OMF Variant V1.0//EN" "http://scrollkeeper.sourceforge.net/dtds/scrollkeeper-omf-1.0/scrollkeeper-omf.dtd">
 <omf>
   <resource>
-   <creator>
-     Ximian, Inc.
-   </creator>
+    <creator>
+      Ximian, Inc.
+    </creator>
     <title>
       Evolution brukerhåndbok
     </title>
     <date>2002</date>
+    <version identifier="" date="2002-10-03"/>
     <subject category="GNOME|Applikasjoner"/>
+    <description/>
+    <type/>
     <format mime="text/sgml"/>
     <identifier url="index.html"/>
     <language code="no"/>
+    <relation seriesid="f98af356-d6f5-11d6-94dd-a55e7b20f7b9"/>
+    <rights type="GNU GPL" license.version="2" license="http://www.gnu.org/licenses/gpl.html"/>    
   </resource>
 </omf>
