--- evolution-1.4.0/acinclude.m4.orig	Mon Apr  7 17:41:13 2003
+++ evolution-1.4.0/acinclude.m4	Thu Jul 10 11:50:10 2003
@@ -129,9 +129,9 @@
 			AC_MSG_ERROR(evolution requires OpenLDAP version >= 2)
 		fi
 
-		AC_CHECK_LIB(resolv, res_query, LDAP_LIBS="-lresolv")
-		AC_CHECK_LIB(socket, bind, LDAP_LIBS="$LDAP_LIBS -lsocket")
-		AC_CHECK_LIB(nsl, gethostbyaddr, LDAP_LIBS="$LDAP_LIBS -lnsl")
+		dnl AC_CHECK_LIB(resolv, res_query, LDAP_LIBS="-lresolv")
+		AC_CHECK_FUNC(bind, , [AC_CHECK_LIB(socket, bind, LDAP_LIBS="$LDAP_LIBS -lsocket")])
+		AC_CHECK_FUNC(gethostbyaddr, , [AC_CHECK_LIB(nsl, gethostbyaddr, LDAP_LIBS="$LDAP_LIBS -lnsl")])
 		AC_CHECK_LIB(lber, ber_get_tag, [
 			if test "$with_static_ldap" = "yes"; then
 				LDAP_LIBS="$with_openldap/lib/liblber.a $LDAP_LIBS"
