diff -urN aa/evolution-1.5.6/camel/camel-debug.c evolution-1.5.6/camel/camel-debug.c
--- aa/evolution-1.5.6/camel/camel-debug.c	2004-04-06 20:33:57.000000000 +0200
+++ evolution-1.5.6/camel/camel-debug.c	2004-04-07 12:42:30.354758088 +0200
@@ -153,8 +153,6 @@
 	pthread_mutex_unlock(&debug_lock);
 }
 
-#include <sys/debugreg.h>
-
 #if 0
 static unsigned
 i386_length_and_rw_bits (int len, enum target_hw_bp_type type)
