--- evolution-2.10.0/composer/Makefile.am.orig	2007-03-25 17:49:13.000000000 +0000
+++ evolution-2.10.0/composer/Makefile.am	2007-03-25 18:37:21.000000000 +0000
@@ -69,6 +69,17 @@
 	listener.c				\
 	listener.h
 
+composerinlcudedir = $(privincludedir)/composer
+composerinlcude_HEADERS = 			\
+	Editor.h				\
+	e-msg-composer.h			\
+	e-msg-composer-hdrs.h
+
+#Composer.h
+#e-msg-composer-select-file.h
+#evolution-composer.h
+#listener.h
+
 EXTRA_DIST =					\
 	mail-composer.error.xml			\
 	$(IDLS)					\
