diff -Nur evolution-1.0.1/configure.in evolution-1.0.1.new/configure.in
--- evolution-1.0.1/configure.in	Wed Dec 19 01:16:23 2001
+++ evolution-1.0.1.new/configure.in	Sun Jan 13 04:46:23 2002
@@ -331,7 +331,7 @@
 	ac_cv_db3_ldadd=""
 
 	for name in db db3 db-3.1; do
-		LIBS="$LIBS_save $with_db3_libs/lib${name}.a"
+		LIBS="$LIBS_save $with_db3_libs/lib${name}.so"
 		AC_TRY_LINK([
 			#ifdef HAVE_DB3_DB_H
 			#include <db3/db.h>
@@ -342,7 +342,7 @@
 			DB *db;
 			db_create (&db, 0, 0);
 		], [
-			ac_cv_db3_ldadd="$with_db3_libs/lib${name}.a"
+			ac_cv_db3_ldadd="$with_db3_libs/lib${name}.so"
 			break
 		])
 	done
