diff -Nur evolution-2.7.90.orig/configure.in evolution-2.7.90.as_needed-fix/configure.in
--- evolution-2.7.90.orig/configure.in	2006-08-06 19:16:24.000000000 +0000
+++ evolution-2.7.90.as_needed-fix/configure.in	2006-08-06 19:18:14.000000000 +0000
@@ -562,8 +562,8 @@
 
 	CFLAGS_save="$CFLAGS"
 	CFLAGS="$CFLAGS $GNOME_PILOT_CFLAGS"
-	LDFLAGS_save="$LDFLAGS"
-	LDFLAGS="$LDFLAGS $GNOME_PILOT_LIBS"
+	LIBS_save="$LIBS"
+	LIBS="$LIBS $GNOME_PILOT_LIBS"
 
 	AC_CACHE_CHECK([if pilot-link handles UTF-8 conversions], ac_cv_pilot_link_utf8, AC_TRY_RUN([
 
@@ -585,7 +585,7 @@
 ], ac_cv_pilot_link_utf8=yes, ac_cv_pilot_link_utf8=no, ac_cv_pilot_link_utf8=no))
 
 	CFLAGS="$CFLAGS_save"
-	LDFLAGS="$LDFLAGS_save"
+	LIBS="$LIBS_save"
 
 	if test "$ac_cv_pilot_link_utf8" = no; then
 		AC_MSG_ERROR(evolution requires pilot-link to have working UTF-8 conversion routines)
