--- evolution-2.22.3.1/widgets/misc/e-unicode.h.orig	2008-04-04 11:17:49.000000000 +0200
+++ evolution-2.22.3.1/widgets/misc/e-unicode.h	2008-09-10 13:36:18.321933761 +0200
@@ -26,7 +26,7 @@
 
 #include <sys/types.h>
 #include <glib.h>
-#include <gtk/gtkclist.h>
+#include <gtk/gtk.h>
 #include <gtk/gtkentry.h>
 #include <gtk/gtkmenu.h>
 #include <gtk/gtkwidget.h>
--- evolution-2.22.3.1/mail/em-account-prefs.h.orig	2008-04-04 11:11:42.000000000 +0200
+++ evolution-2.22.3.1/mail/em-account-prefs.h	2008-09-10 14:12:31.254765728 +0200
@@ -31,7 +31,7 @@
 
 #include <gtk/gtkvbox.h>
 #include <gtk/gtkbutton.h>
-#include <gtk/gtkclist.h>
+#include <gtk/gtk.h>
 #include <glade/glade.h>
 #include <gtk/gtktreeview.h>
 
