diff -urN aaa/evolution-1.4.5/evolution-addressbook.pc.in evolution-1.4.5/evolution-addressbook.pc.in
--- aaa/evolution-1.4.5/evolution-addressbook.pc.in	2003-05-22 22:18:00.000000000 +0200
+++ evolution-1.4.5/evolution-addressbook.pc.in	2003-11-20 14:50:13.000000000 +0100
@@ -13,5 +13,5 @@
 Description: libraries needed for Evolution addressbook backends
 Version: @VERSION@
 Requires: gconf-2.0 libbonobo-2.0 gal-2.0 >= @GAL_REQUIRED@ libgnome-2.0 camel = @VERSION@
-Libs: -L${privlibdir} -lpas -lebook -lename -lversit
+Libs: -L${privlibdir} -lebook -lename -lversit
 Cflags: -I${privincludedir}
diff -urN aaa/evolution-1.4.5/evolution-calendar.pc.in evolution-1.4.5/evolution-calendar.pc.in
--- aaa/evolution-1.4.5/evolution-calendar.pc.in	2003-05-22 22:18:00.000000000 +0200
+++ evolution-1.4.5/evolution-calendar.pc.in	2003-11-20 14:50:20.000000000 +0100
@@ -13,5 +13,5 @@
 Description: libraries needed for Evolution calendar backends
 Version: @VERSION@
 Requires: libgnome-2.0 libbonobo-2.0 gal-2.0 >= @GAL_REQUIRED@ gnome-vfs-2.0
-Libs: -L${privlibdir} -lpcs -lcal-util -lcal-client -lical-evolution -lwombat -leutil
+Libs: -L${privlibdir} -lcal-util -lcal-client -lical-evolution -lwombat -leutil
 Cflags: -I${privincludedir}
