diff -urN evolution-2.29.92.1/configure.ac evolution-2.29.92.1.new//configure.ac
--- evolution-2.29.92.1/configure.ac	2010-03-10 14:49:07.000000000 +0100
+++ evolution-2.29.92.1.new//configure.ac	2010-03-13 23:02:43.212014431 +0100
@@ -537,11 +537,6 @@
 AC_CHECK_FUNCS(statfs)
 
 dnl **************************************************
-dnl Gnome Icon Theme
-dnl **************************************************
-PKG_CHECK_MODULES([GIT], [gnome-icon-theme >= gnome_icon_theme_minimum_version])
-
-dnl **************************************************
 dnl Accessibility support
 dnl **************************************************
 PKG_CHECK_MODULES([A11Y], [atk])
